api/v1alpha1/flowcollector_webhook.go

```diff
@@ -20,63 +20,75 @@ import (
 	"fmt"
 	"reflect"
 
-	"github.com/netobserv/network-observability-operator/api/v1beta1"
+	"github.com/netobserv/network-observability-operator/api/v1beta2"
 	utilconversion "github.com/netobserv/network-observability-operator/pkg/conversion"
 	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	apiconversion "k8s.io/apimachinery/pkg/conversion"
 	"sigs.k8s.io/controller-runtime/pkg/conversion"
 )
 
-// ConvertTo converts this v1alpha1 FlowCollector to its v1beta1 equivalent (the conversion Hub)
+// ConvertTo converts this v1alpha1 FlowCollector to its v1beta2 equivalent (the conversion Hub)
 // https://book.kubebuilder.io/multiversion-tutorial/conversion.html
 func (r *FlowCollector) ConvertTo(dstRaw conversion.Hub) error {
-	dst := dstRaw.(*v1beta1.FlowCollector)
+	dst := dstRaw.(*v1beta2.FlowCollector)
 
-	if err := Convert_v1alpha1_FlowCollector_To_v1beta1_FlowCollector(r, dst, nil); err != nil {
-		return fmt.Errorf("copying v1alpha1.FlowCollector into v1beta1.FlowCollector: %w", err)
+	if err := Convert_v1alpha1_FlowCollector_To_v1beta2_FlowCollector(r, dst, nil); err != nil {
+		return fmt.Errorf("copying v1alpha1.FlowCollector into v1beta2.FlowCollector: %w", err)
 	}
 	dst.Status.Conditions = make([]v1.Condition, len(r.Status.Conditions))
 	copy(dst.Status.Conditions, r.Status.Conditions)
 
 	// Manually restore data.
-	restored := &v1beta1.FlowCollector{}
+	restored := &v1beta2.FlowCollector{}
 	if ok, err := utilconversion.UnmarshalData(r, restored); err != nil || !ok {
+		// fallback on current loki config as Manual mode if metadata are not available
+		dst.Spec.Loki.Mode = v1beta2.LokiModeManual
+		dst.Spec.Loki.Manual.IngesterURL = r.Spec.Loki.URL
+		dst.Spec.Loki.Manual.QuerierURL = r.Spec.Loki.QuerierURL
+		dst.Spec.Loki.Manual.StatusURL = r.Spec.Loki.StatusURL
+		dst.Spec.Loki.Manual.TenantID = r.Spec.Loki.TenantID
+		dst.Spec.Loki.Manual.AuthToken = r.Spec.Loki.AuthToken
+		if err := Convert_v1alpha1_ClientTLS_To_v1beta2_ClientTLS(&r.Spec.Loki.TLS, &dst.Spec.Loki.Manual.TLS, nil); err != nil {
+			return fmt.Errorf("copying v1alplha1.Loki.TLS into v1beta2.Loki.Manual.TLS: %w", err)
+		}
```

Comment:
I had issues when the storage was on `v1beta2` :thinking: 
Let's see if I can remove these now that the storage is on `v1beta1`

Reply:
I confirm we still need these pieces for the first deployment.
You can reproduce by deploying the operator and applying `flows_v1alpha1_flowcollector.yaml` with a custom loki url from scratch (no existing CR instance).

Without this part of code we lost the custom url and it fallback to its default since the field was moved from `spec.loki.url` to `spec.loki.manual.url`.